esp/src/src-react/hooks/user.ts

```diff
@@ -1,6 +1,67 @@
 import * as React from "react";
-import { useConst } from "@fluentui/react-hooks";
+import { useConst, useForceUpdate } from "@fluentui/react-hooks";
 import { AccountService, WsAccount } from "@hpcc-js/comms";
+import { userKeyValStore } from "src/KeyValStore";
+
+const defaults = {
+    ESPSessionTimeout: 7200,
+    ESPAuthenticated: false,
+    ECLWatchUser: false,
+    Status: "Unlocked",
+    ESPSessionState: false
+};
+
+const session = { ...defaults };
+
+export interface UserSession {
+    ESPSessionTimeout: number;
+    ESPAuthenticated: boolean;
+    ECLWatchUser: boolean;
+    Status: string;
+    ESPAuthenticationMSG?: string;
+    ESPSessionState: boolean;
+}
+
+export function useUserSession(): [UserSession, (opts: UserSession) => void, () => void, (toDefaults?: boolean) => void, () => Promise<void>] {
+
+    const store = useConst(() => userKeyValStore());
+    const refresh = useForceUpdate();
+
+    const setSession = React.useCallback((opts: UserSession) => {
+        for (const key in opts) {
+            session[key] = opts[key];
+        }
+        refresh();
+    }, [refresh]);
```

Comment:
Does refresh here, trigger all pages that have a useUserSession instance to re-render?